src/Qbead.h

```diff
@@ -221,31 +332,68 @@ class Qbead {
     leg = nlegs - leg; // invert direction for the phi angle, because the PCB is set up as a left-handed coordinate system
     leg = leg % nlegs;
     if (leg == 0) {
-      pixels.setPixelColor(pixel, color);
+        pixels.setPixelColor(pixel, color);
     } else if (pixel == 0) {
-      pixels.setPixelColor(0, color);
+        pixels.setPixelColor(0, color);
     } else if (pixel == 6) {
-      pixels.setPixelColor(6, color);
+        pixels.setPixelColor(6, color);
     } else {
-      pixels.setPixelColor(7 + (leg - 1) * (nsections - 1) + pixel - 1, color);
+        pixels.setPixelColor(7 + (leg - 1) * (nsections - 1) + pixel - 1, color);
     }
   }
 
   void setBrightness(uint8_t b) {
     pixels.setBrightness(b);
   }
 
+  void readIMU() {
+    rbuffer[0] = imu.readFloatAccelX();
+    rbuffer[1] = imu.readFloatAccelY();
+    rbuffer[2] = imu.readFloatAccelZ();
+    rx = (1 - 2 * sx) * rbuffer[ix];
+    ry = (1 - 2 * sy) * rbuffer[iy];
+    rz = (1 - 2 * sz) * rbuffer[iz];
+
+    float T_new = micros();
+    float delta = T_new - T_imu;
+    T_imu = T_new;
+    const float T = 100000; // 100 ms // TODO make the filter timeconstant configurable
+    if (delta > 100000) {
+        state.setXYZ(rx, ry, rz);
+    } else {
+        float d = delta / T;
+        float newX = d * rx + (1 - d) * state.getX();
+        float newY = d * ry + (1 - d) * state.getY();
+        float newZ = d * rz + (1 - d) * state.getZ();
+        state.setXYZ(newX, newY, newZ);
+    }
+
+    state.printState();
+
+    rbuffer[0] = state.getX();
+    rbuffer[1] = state.getY();
+    rbuffer[2] = state.getZ();
+    blecharacc.write(rbuffer, 3*sizeof(float));
+    for (uint16_t conn_hdl=0; conn_hdl < QB_MAX_PRPH_CONNECTION; conn_hdl++)
+    {
+      if ( Bluefruit.connected(conn_hdl) && blecharacc.notifyEnabled(conn_hdl) )
+      {
+        blecharacc.notify(rbuffer, 3*sizeof(float));
+      }
+    }
+  }
```

Comment:
Any reason for moving this function?

Generally I prefer to keep cosmetic changes separate so that we can add them to the list of commits that git-blame ignores (in `.git-blame-ignore-revs`). Otherwise down the line it becomes pretty difficult to use git log and git blame to figure out why a change was made.

It also makes it easier to review as it is more obvious if there were changes to the function.